### PR TITLE
Create CONTRIBUTING and some templates

### PR DESCRIPTION
**Pull request by colindean**

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/new_check.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+---
+
+# What would you like to check?
+
+<!-- Provide as much detail as you can about what precisely you want to check. -->
+
+# What does the configuration look like?
+
+<!-- 
+Provide an example of how you would like the configuration for the check to look.
```
Comment (phpisciuneri) [inline .github/ISSUE_TEMPLATE/new_check.md:17]:
```suggestion
Provide documentation of how you envision the configuration of the validator to function.  Please follow the style of other validators in the README, see https://github.com/target/data-validator#validators.
```

File: .github/PULL_REQUEST_TEMPLATE/new_check.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+# What issue is this PR solving?
+
+<!-- Provide the link. -->
+
+# What does the configuration look like?
+
+<!-- 
+Provide an example of how you would like the configuration for the check to look. 
+Most of our rework requests are the result of an unclear vision for the interface to the check!
```
Comment (phpisciuneri) [inline .github/PULL_REQUEST_TEMPLATE/new_check.md:18]:
```suggestion
```

File: .github/PULL_REQUEST_TEMPLATE/new_check.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+---
+
+# What issue is this PR solving?
+
+<!-- Provide the link. -->
+
+# What does the configuration look like?
+
+<!-- 
+Provide an example of how you would like the configuration for the check to look.
```
Comment (phpisciuneri) [inline .github/PULL_REQUEST_TEMPLATE/new_check.md:17]:
```suggestion
#### validatorName

Provide a brief description of check and the condition that will result in a failure.  Document any arguments in the table below.
 
| Arg | Type | Description |
|-----|------|-------------| 
```

File: .github/PULL_REQUEST_TEMPLATE/new_check.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+# What does the configuration look like?
+
+<!-- 
+Provide an example of how you would like the configuration for the check to look. 
+Most of our rework requests are the result of an unclear vision for the interface to the check!
+-->
+
+# Have you completed all of these?
+
+- [ ] Pass the style checker requirements without warnings or errors (`sbt test` will not work without compliance!)
```
Comment (phpisciuneri) [inline .github/PULL_REQUEST_TEMPLATE/new_check.md:23]:
```suggestion
- [ ] Add configuration documentation to the Validators section of the README.  You should be able to copy this from the previous section.
- [ ] Pass the style checker requirements without warnings or errors (`sbt test` will not work without compliance!)
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        1. Configuration parsing
+        2. Configuration sanity checking
+        3. Variable substitution
+        4. Actual check functionality
+
+## Refactoring
+
+Follow the new checks procedure, but instead of providing a configuration example, clearly explain:
+
+1. How the current state of things negatively affects you or encumbers your use of Data Validator
```
Comment (phpisciuneri) [inline CONTRIBUTING.md:29]:
```suggestion
1. How the current state of things negatively affects the extensibility of Data Validator.
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+2. How you intend to remedy the situation with the minimum amount of code changed
```
Comment (phpisciuneri) [inline CONTRIBUTING.md:30]:
```suggestion
2. How you intend to remedy the situation with the minimum amount of code changed

**Do not mix refactoring with the addition of a new check in the same pull request.** We will reject and ask that they be done in separate PRs to keep things manageable.
```

File: .github/PULL_REQUEST_TEMPLATE/new_check.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+- [ ] Include tests. Submissions without tests will not be considered. Test the following things:
+     - [ ] Configuration parsing
+     - [ ] Configuration sanity checking
+     - [ ] Variable substitution
+     - [ ] Actual check functionality
```
Comment (phpisciuneri) [inline .github/PULL_REQUEST_TEMPLATE/new_check.md:28]:
```suggestion
     - [ ] Actual check functionality
     
**Important:** Your submission should not modify any of the other validators.  Please see our note on Refactoring in the CONTRIBUTING doc.
```

File: .github/PULL_REQUEST_TEMPLATE/new_check.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+<!-- 
+Provide an example of how you would like the configuration for the check to look. 
+Most of our rework requests are the result of an unclear vision for the interface to the check!
+-->
+
+# Have you completed all of these?
+
+- [ ] Pass the style checker requirements without warnings or errors (`sbt test` will not work without compliance!)
+- [ ] Include tests. Submissions without tests will not be considered. Test the following things:
```
Comment (colindean) [inline .github/PULL_REQUEST_TEMPLATE/new_check.md:24]:
```suggestion
- [ ] Does not modify any of the other validators. Please review the section Refactoring in the CONTRIBUTING.md.
- [ ] Include tests. Submissions without tests will not be considered. Test the following things:
```

File: .github/PULL_REQUEST_TEMPLATE/new_check.md (hunk 7 above)
Comment (colindean) [reply .github/PULL_REQUEST_TEMPLATE/new_check.md:28]: Making this a checkbox instead of a note. Minimally, it's a gimme.

**Pull request merged**